File: android/src/main/java/top/huic/tencent_im_plugin/message/FaceMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.v2.V2TIMFaceElem;
import com.tencent.imsdk.v2.V2TIMManager;
import com.tencent.imsdk.v2.V2TIMMessage;

import top.huic.tencent_im_plugin.message.entity.FaceMessageEntity;

/**
 * 表情消息节点
 */
public class FaceMessageNode extends AbstractMessageNode<V2TIMFaceElem, FaceMessageEntity> {

    @Override
    public V2TIMMessage getV2TIMMessage(FaceMessageEntity entity) {
        return V2TIMManager.getMessageManager().createFaceMessage(entity.getIndex(), entity.getData().getBytes());
    }

    @Override
    public String getNote(V2TIMFaceElem elem) {
        return "[表情]";
    }

    @Override
    public FaceMessageEntity analysis(V2TIMFaceElem elem) {
        return new FaceMessageEntity(elem);
    }

    @Override
    public Class<FaceMessageEntity> getEntityClass() {
        return FaceMessageEntity.class;
    }
}

File: ios/Classes/message/FaceMessageNode.swift
//
// Created by 蒋具宏 on 2020/10/27.
//

import Foundation
import ImSDK

/// 表情消息实体
class FaceMessageNode: AbstractMessageNode {
    override func getV2TIMMessage(params: [String: Any]) -> V2TIMMessage {
        let index: Int32 = getParam(params: params, paramKey: "index")!;
        let data: String = getParam(params: params, paramKey: "data")!;
        return V2TIMManager.sharedInstance().createFaceMessage(index, data: data.data(using: String.Encoding.utf8))
    }

    override func getNote(elem: V2TIMElem) -> String {
        "[表情]";
    }

    override func analysis(elem: V2TIMElem) -> AbstractMessageEntity {
        FaceMessageEntity(elem: elem as! V2TIMFaceElem)
    }
}
